static/src/javascripts/projects/common/modules/audio/inArticlePlayer.js

```diff
@@ -0,0 +1,135 @@
+};
+
+// design hack to give immediate feedback on the progress bar for a play event
+const showStarterBlockOnFirstPlay = () => {
+    const scrubberBar: ?HTMLElement = document.querySelector(
+        '.inline-audio_content_progress-bar'
+    );
+    if (scrubberBar) scrubberBar.classList.add('started');
+};
+
+// PLAYER ACTION FUNCTIONS
+
+const updateTime = (el: HTMLElement, player, progressBar: HTMLElement) => {
+    let prevTime = 0;
+    player.addEventListener('timeupdate', () => {
+        el.textContent = formatTime(Math.round(player.currentTime));
+        const percentPlayed = (player.currentTime / player.duration) * 100;
+        // throttle updates to once a second
+        if (Math.round(player.currentTime) !== Math.round(prevTime)) {
+            fastdom.write(() => {
+                progressBar.style.width = `${percentPlayed}%`;
+            });
+            prevTime = player.currentTime;
+        }
+    });
+};
+
+const activateAudioControls = (el, player, id) => {
+    el.addEventListener('click', () => {
+        if (player.paused) {
+            sendToOphan(id, 'play');
+            player.play();
+            setPauseButton(el);
+            showStarterBlockOnFirstPlay();
+        } else {
+            player.pause();
+            setPlayButton(el);
+        }
+    });
+};
+
+const activateScrubber = (el, player) => {
+    const w = el.offsetWidth;
+    const d = player.duration;
+
+    el.addEventListener('click', (e: MouseEvent | Touch) => {
+        const leftOffset = el.getBoundingClientRect().left;
+        const clickX = e.clientX - leftOffset;
+        const ratio = clickX / w;
+        player.currentTime = Math.round(ratio * d);
+    });
+};
+
+const setDuration = (el, duration) => {
+    el.textContent = formatTime(Math.round(duration));
+};
+
+// INITIALISER
+const init = () => {
+    const player = document.querySelector('audio.inline-audio-player-element');
+
+    if (player && !(player instanceof HTMLMediaElement)) return;
+
+    const container: ?HTMLElement = document.querySelector(
+        '.inline-audio_container'
+    );
+
+    const mediaId: ?string =
+        player && player.hasAttribute('data-media-id')
+            ? player.getAttribute('data-media-id')
+            : '';
+    const buttonDiv: ?HTMLElement = document.querySelector(
+        '.inline-audio_button'
+    );
+    const scrubberBar: ?HTMLElement = document.querySelector(
+        '.inline-audio_content_progress-bar'
+    );
+    const progressBar: ?HTMLElement = document.querySelector(
+        '.inline-audio_content_progress-bar .played'
+    );
+    const timePlayedSpan: ?HTMLElement = document.querySelector(
+        '#inline-audio_time-played'
+    );
+    const durationSpan: ?HTMLElement = document.querySelector(
+        '#inline-audio_duration'
+    );
+
+    if (
+        player &&
+        mediaId &&
+        buttonDiv &&
+        scrubberBar &&
+        progressBar &&
+        timePlayedSpan &&
+        durationSpan
+    ) {
```

Comment:
Inside that `if`, Flow should know that all the vars above have a value, meaning you could update functions like `playerObserved` to
```
const playerObserved = (el: HTMLElement, id: string) => {
  // no need to test `el` here
  ...
}
```

it's a minor thing that can wait for when this code is turned into production code.

Reply:
ah.. the benefits of flow..

Reply:
Actually flow wouldn't let me do that :( maybe the flow checker doesn't carry its knowledge between files?